vinhloc30796/creativecontact-app
# feat(i18n): fix language detection middleware & profile translations

## Description

Implements a comprehensive internationalization (i18n) system with language detection, persistence, and translations for profile components. Includes middleware improvements and client-side optimizations.

## Key Changes

0. Fixed infinite loop
1. Implemented new language detection middleware with path, cookie, and header support
2. Added translations for ProfileCard and portfolio sections (EN/VI)
3. Converted server components to client-side with React Query integration
4. Added placeholder image system for profile pictures
5. Improved language persistence with cookie-based storage

## Breaking Changes

- Language detection now follows strict priority: path > cookie > header
- Changed profile component props to use Promises instead of direct data
- Removed direct URL-based language switching in favor of cookie-based persistence

## Related Issues

- Closes WEB-249

## Testing Instructions

1. Change language using different methods:
   - URL parameter (?lang=vi)
   - Browser language settings
   - Cookie settings

2. Verify profile components:
   - Check translations in both English and Vietnamese
   - Verify placeholder images for missing profile pictures
   - Test social media links and portfolio display

3. Test language persistence:
   - Switch between pages
   - Refresh browser
   - Clear cookies and verify fallback behavior

## Additional Notes

- Added new translation files in `/public/translations/`
- Implemented caching for translation hooks
- Added skeleton loading states for profile components
- DiceBear is now the default placeholder image provider
